### PR TITLE
Ntrnl 452 add get user email method from login module to the GitHub request app and update the notify call to be dynamic

### PR DESCRIPTION
**Pull request by DanielMurray97**
### JIRA link

Resolves [NTRNL-452](https://technologyprogramme.atlassian.net/browse/NTRNL-452?atlOrigin=eyJpIjoiYjA1ZmY5YTlhNWFkNGNjYjgxZDQ1ZTY4NWQ3NTRjNDUiLCJwIjoiaiJ9)

### Description

- Integrate `getUserEmailFromColaJwt` method from node-login module
- Create util `getUserEmail` using `getUserEmailFromColaJwt` to collect the users email.
- Change email in `confirmationEmail` to be dynamic
- Updated tests based on the implementation of `getUserEmail`

Screenshot from 2024-05-29 15-55-55

### Work checklist

- [x] Tests added where applicable
- [x] No vulnerability added

[NTRNL-452]: https://technologyprogramme.atlassian.net/browse/NTRNL-452?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by harley-harris** [comment]
Pedantic but could `FEATURE_FLAG_NOTIFY` be updated to `FEATURE_FLAG_ENABLE_NOTIFY` for naming consistency in `.env.example`?

```
FEATURE_FLAG_ENABLE_AUTH="true"
FEATURE_FLAG_ENABLE_NOTIFY="true"
```

File: src/utils/getUserEmail.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { getUserEmailFromColaJwt } from '@co-digital/login';
+import { log } from '../utils/logger';
+import { Request } from 'express';
+
+import * as config from '../config';
+
+export const getUserEmail = (req: Request) => {
+    if (config.NODE_ENV === 'production') {
+        const jwt = req.signedCookies[config.COOKIE_ID_NAME];
```
Comment (Mouhajer-CO) [inline src/utils/getUserEmail.ts:9]: We should not handle the cookies directly in the app. Let's go with it for now, but we will add a new utility on   `@co-digital/login` module

File: src/controller/check-your-requests.controller.ts [hunk 2]
```diff
@@ -40,8 +41,10 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
 
         await client.gitHub.postIssue(url, body);
 
-        // TODO: Implement logic to gather the users email
-        await confirmationEmail(config.NOTIFY_USER_EMAIL, id);
+        const userEmail = getUserEmail(req);
+        if (userEmail) {
```
Comment (Mouhajer-CO) [inline src/controller/check-your-requests.controller.ts:45]: This check is always true based on the control in the getUserEmail(...) method. I would suggest handling the logic in the confirmationEmail(...) method.
I would have just remove the comment here.

File: src/service/notify.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import * as config from '../config';
 
 export const confirmationEmail = async (emailAddress: string, reference_number: string): Promise<void> => {
```
Comment (Mouhajer-CO) [inline src/service/notify.ts:6]: To reduce complexity and remove duplication I would have added getUserEmail method here and remove the need to add logic on the check-your-answer page and on the getUserEmail util, something like:
```suggestion
export const confirmationEmail = async (jwt: string, reference_number: string): Promise<void> => {
    if (isFeatureEnabled(config.FEATURE_FLAG_ENABLE_NOTIFY)) {
        const notifyClient = new NotifyClient(config.NOTIFY_API_KEY);
        const emailAddress = getUserEmail(jwt);

        log.info(`Email confirmation is enabled. User: ${emailAddress}.`);

        await notifyClient.sendEmail(config.NOTIFY_EMAIL_TEMPLATE, emailAddress, {
           ...
        });
    } else {
        log.info('Email confirmation is disabled.');
    }
};
```

File: src/utils/getUserEmail.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { getUserEmailFromColaJwt } from '@co-digital/login';
+import { log } from '../utils/logger';
+import { Request } from 'express';
+
+import * as config from '../config';
+
+export const getUserEmail = (req: Request) => {
```
Comment (Mouhajer-CO) [inline src/utils/getUserEmail.ts:7]: I would have changed it to

```suggestion
export const getUserEmail = (jwt: string): string => {
    const userEmail = getUserEmailFromColaJwt(jwt);

    if (!userEmail) {
        throw new Error('Failed to fetch the email.');
    }

    return userEmail;
};
```
Furthermore, we are going to use this method to fetch user's email and add it to the header when logging in.

**Review by Mouhajer-CO — COMMENTED** [review]
I have noticed that we need to change the email address added to the notification template. It should be similar to the contact information on the confirmation page instead of the user itself

File: src/utils/getUserEmail.ts (hunk 1 above)
Comment (Mouhajer-CO) [reply src/utils/getUserEmail.ts]: Thanks @DanielMurray97

**Comment by Mouhajer-CO** [comment]
> I have noticed that we need to change the email address added to the notification template. It should be similar to the contact information on the confirmation page instead of the user itself

As suggested by @DanielMurray97, the email is set as an environment variable and is already customisable

**Review by Mouhajer-CO — APPROVED** [review]
Awesome @DanielMurray97 , thanks 🙏

**Pull request merged**